packages/mcp-server-supabase/src/tools/branching-tools.ts

```diff
@@ -37,25 +37,73 @@ export function getBranchingTools({
           .string()
           .default('develop')
           .describe('Name of the branch to create'),
+        // When the client supports elicitation, we will ask the user to confirm the
+        // branch cost interactively and this parameter is not required. For clients
+        // without elicitation support, this confirmation ID is required.
         confirm_cost_id: z
-          .string({
-            required_error:
-              'User must confirm understanding of costs before creating a branch.',
-          })
-          .describe('The cost confirmation ID. Call `confirm_cost` first.'),
+          .string()
+          .optional()
```

Comment:
Wonder if we can make the schema / tool implementation dynamic? For clients that don't support elicitation, the field isn't really optional.